mypyc/ir/rtypes.py

```diff
@@ -174,7 +174,7 @@ def __init__(self,
         self.is_unboxed = is_unboxed
         self._ctype = ctype
         self.is_refcounted = is_refcounted
-        if ctype == 'CPyTagged':
+        if ctype in {'CPyTagged', 'int'}:
```

Comment:
Minor performance suggestion: `x in (y, z)` is generally faster than `x in {y, z}` (at least when there are only a few items in the set).